src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -580,25 +586,29 @@ public class MultipleMutationInputValidator
         /// In case when a value for referencing column is also specified in the referencing entity, there can be two conflicting sources of truth,
         /// which we don't want to allow. In such a case, we throw an appropriate exception.
         /// </summary>
-        /// <param name="columnsInTargetEntity">Columns in the target (referencing) entity.</param>
+        /// <param name="backingColumnToSourceMapInTargetEntity">Dictionary storing mapping from names of backing columns in the target (referencing) entity to the source
+        /// of the corresponding column's value The source can either be:
+        /// 1. User provided scalar input for a column
+        /// 2. Insertion in the source (referenced) entity.</param>
```

Comment:
nit: `backingColumnToSourceMapInTargetEntity` I'm not clear on this naming.. 
do you mean `columnNameToSourcedValue`? can you give possible values for a dictionary value?

Reply:
Added comment to both the cases' source values.